libraries/apollo-ast/src/test/kotlin/com/apollographql/apollo3/graphql/ast/test/LexerTest.kt

```diff
@@ -0,0 +1,177 @@
+package com.apollographql.apollo3.graphql.ast.test
+
+import com.apollographql.apollo3.ast.builtinDefinitions
+import com.apollographql.apollo3.ast.internal.Lexer
+import com.apollographql.apollo3.ast.internal.LexerException
+import com.apollographql.apollo3.ast.internal.Token
+import okio.Buffer
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+
+/**
+ * Test ported from graphql-js
+ * https://github.com/graphql/graphql-js/blob/8724e22f9e89224dd2669d6fc0c4b7fe34c2c8dd/src/language/__tests__/lexer-test.ts
+ */
+class LexerTest {
+  private fun scanFirst(string: String): Token {
+    return Lexer(Buffer().writeUtf8(string)).run {
+      nextToken()
+      nextToken()
+    }
+  }
+
+  private fun scanSecond(string: String): Token {
+    return Lexer(Buffer().writeUtf8(string)).run {
+      nextToken()
+      nextToken()
+      nextToken()
+    }
+  }
+
+  private fun Token.assertName(value: String) {
+    assertIs<Token.Name>(this)
+    assertEquals(value, this.value)
+  }
+
+  private fun Token.assertString(value: String) {
+    assertIs<Token.String>(this)
+    assertEquals(value, this.value)
+  }
+
+  @Test
+  fun ignoresBOMHeader() {
+    scanFirst("\uFEFF foo").apply {
+      assertName("foo")
+    }
+  }
+
+  @Test
+  fun tracksLineBreaks() {
+    scanFirst("foo").apply {
+      assertName("foo")
+      assertEquals(1, line)
+      assertEquals(1, column)
+    }
+    scanFirst("\nfoo").apply {
+      assertName("foo")
+      assertEquals(2, line)
+      assertEquals(1, column)
+    }
+    scanFirst("\n\rfoo").apply {
+      assertName("foo")
+      assertEquals(3, line)
+      assertEquals(1, column)
+    }
+    scanFirst("\r\r\n\nfoo").apply {
+      assertName("foo")
+      assertEquals(4, line)
+      assertEquals(1, column)
+    }
+    scanFirst("\n\n\r\rfoo").apply {
+      assertName("foo")
+      assertEquals(5, line)
+      assertEquals(1, column)
+    }
+  }
+
+  @Test
+  fun tracksLineAndColumn() {
+    scanFirst("\n \r\n \r  foo\n").apply {
+      assertName("foo")
+      assertEquals(4, line)
+      assertEquals(3, column)
+    }
+  }
+
+  @Test
+  fun skipsWhitespaceAndComments() {
+    scanFirst(
+        """
+
+              foo
+
+
+              """
+    ).apply {
+      assertName("foo")
+    }
+
+    scanFirst("\t\tfoo\t\t").apply {
+      assertName("foo")
+    }
+
+    scanFirst("""
+      #comment
+      foo#comment
+    """).apply {
+      assertName("foo")
+    }
+
+    scanFirst(",,,foo,,,").apply {
+      assertName("foo")
+    }
+  }
+
+  @Test
+  fun lexesBlockString() {
+    builtinDefinitions()
```

Comment:
I'm confused by this one

Reply:
It's something I wrote during dev to try out something but it's not relevant indeed. Removed, thanks!